.github/workflows/preview.yml

```diff
@@ -0,0 +1,112 @@
+name: 'Preview'
+on:
+  pull_request:
+    types:
+      - synchronize
+      - opened
+      - closed
+
+env:
+  BRANCH: ${{ github.event.pull_request.head.ref }}
+  CC_CLEVER_TOOLS_PREVIEWS_CELLAR_KEY_ID: ${{ secrets.CC_CLEVER_TOOLS_PREVIEWS_CELLAR_KEY_ID }}
+  CC_CLEVER_TOOLS_PREVIEWS_CELLAR_SECRET_KEY: ${{ secrets.CC_CLEVER_TOOLS_PREVIEWS_CELLAR_SECRET_KEY }}
+
+jobs:
+  wait_for_build:
+    name: 'Wait for build to succeed'
+    if: |
+      (github.event.action == 'synchronize' || github.event.action == 'opened') &&
+      github.event.repository.fork == false &&
+      !startsWith(github.event.pull_request.head.ref, 'release-please--')
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Wait for build to succeed'
+        uses: lewagon/wait-on-check-action@v1.3.1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          check-name: Build
+          ref: ${{ github.event.pull_request.head.sha }}
+          allowed-conclusions: success
+  publish:
+    name: 'Publish Preview'
+    needs: wait_for_build
+    runs-on: ubuntu-latest
+    steps:
+      - name: '[Prepare] Checkout'
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: '[Prepare] Setup Node.js'
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: 'package.json'
+          cache: 'npm'
+      - name: '[Prepare] Install dependencies'
+        run: npm ci
+      - name: '[Prepare] Check if a preview already exists'
+        id: check_preview
+        run: |
+          if node scripts/job-preview.js get ${BRANCH} > /dev/null 2>&1; then
+            echo "exists=yes" >> $GITHUB_OUTPUT
+          else
+            echo "exists=no" >> $GITHUB_OUTPUT
+          fi
+
+      - name: '[Run] Build preview'
+        run: node scripts/job-preview.js build ${BRANCH}
```

Comment:
${BRANCH} is an env var that was setup at the beginning of the process (line 10). This env is injected in every steps.
I prefer setting BRANCH once, because it make the workflow easier to read.